static/src/javascripts/projects/common/modules/navigation/newSticky.js

```diff
@@ -28,68 +28,109 @@ define([
-                        return iframeHeightStr ? newAdHeight : oldAdHeight;
+                        return $adBannerInner[0].clientHeight;
                     });
                 };
 
-                var previousAdHeight = oldAdHeight;
-                var render = function (options) {
-                    var firstRender = options && options.firstRender;
-                    getLatestAdHeight().then(function (adHeight) {
-                        fastdom.read(function () {
-                            var scrollY = window.scrollY;
-                            var scrollX = window.scrollX;
-
-                            fastdom.write(function () {
-                                // Reset
-                                $header.css('transition', '');
-                                $adBanner.css('top', '');
-
-                                // Set
-                                if (scrollY > headerHeight) {
-                                    $adBanner.css({
-                                        'position': 'absolute',
-                                        'top': headerHeight + 'px'
-                                    });
-                                } else {
-                                    $adBanner.css({ 'position': 'fixed' });
-                                }
-
-                                $header.css('margin-top', adHeight + 'px');
-
-                                var scrollIsAtTop = scrollY === 0;
-                                // Avoid an initial transition when we apply the margin top for the first time
-                                if (scrollIsAtTop && !firstRender) {
-                                    // If the user is at the top of the page, we want to transition
-                                    // the change
-                                    $header.css('transition', 'margin-top 0.75s cubic-bezier(0, 0, 0, 0.985)');
-                                } else {
-                                    // If the user is not at the top, we want to offset their scroll position
-                                    var diff = adHeight - previousAdHeight;
-                                    if (diff > 0) {
-                                        window.scrollTo(scrollX, scrollY + diff);
-                                    }
-                                }
-
-                                previousAdHeight = adHeight;
+                var getLatestAdHeight = function () {
+                    var $iframe = $('iframe', $adBanner);
+                    var slotWidth = $iframe.attr('width');
+                    var slotHeight = $iframe.attr('height');
+                    // iframe may not have been injected at this point
+                    var isFluidAd = $iframe.length > 0 && [slotWidth, slotHeight].join(',') === '88,70';
+                    // fluid ads are currently always 250px high
+                    var fluidAdInnerHeight = 250;
```

Comment:
I think this needs a comment to say why we can't just calculate the client height. We can't calculate the client height because fluid ads are rendered asynchronously.